### ISSUE TITLE
Excavators try to refuel at locations they can't reach 

### ISSUE THREAD

**Issue by siepkes** [MaFi-Games/Captain-of-Industry-issues#597]
**The issue**

Excavators sometimes try to refuel (and get stuck waiting) at locations they can't reach due to for example pipes. This situation can be seen in the attached screenshot. The mining tower is actually connected to the fuel station in the bottom right corner of the screenshot but excavators don't seem to adhere to such a connection (I guess only trucks do)? 

This issue might be related to #593.

**Expected behavior**

I would expect vehicles to prioritize refueling destinations they can reach.

**To Reproduce**

Can provide savegame if needed.

**Additional context**
 - Game version: 

**Attachments**

![captain](https://user-images.githubusercontent.com/1428678/179421780-9f948908-a6fc-4b5d-8017-e7d6f8c4fbab.PNG)

**Comment by zuff-coi** [comment]
If you could provide a save file that would be helpful in figuring out this behavior. Thanks!

**Comment by MaFi-Games** [comment]
Yes, excavators don't recognize assigned fuel stations.

**Comment by siepkes** [comment]
Here it is: [game.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/9214129/game.zip)

**Comment by mloftis** [comment]
I'll have to try to build a repro map, but had an issue related to this.  Excavator selected a fluid storage that became or was inaccessible before it got moving there.  Excavator was then stuck and out of fuel right on the mine until I recovered it even after building a fuel station (fuel station was in progress waiting on Construction II mats).

I have not yet tried it but it does seem timing sensitive.  Have a fuel fluid storage tank that is accessible, but is made inaccessible right around when the excavator is pathing (but maybe not driving for) the fluid storage.  Excavator runs itself out of fuel, never switching to any other accessible fluid storage (there was at least one other at the time this happened).  I had started building a fuel station, let that finish, but it never called for the fuel truck even after a very long time, just stayed stuck inaccessible (even though it was also out of fuel at this point).

I'm not 100% clear on the exact order of events because I only saw the excavator after it was stuck without access to the fluid storage it had decided on going to.  It never tried to go to another one as far as I can tell (And the whole time there was ~40+ fuel sitting in the dock which was accessible even though I was messing around with my tank farm)
